nats.go

```diff
@@ -1450,6 +1450,7 @@ func (nc *Conn) connect() error {
 	// For first connect we walk all servers in the pool and try
 	// to connect immediately.
 	nc.mu.Lock()
+	defer nc.mu.Unlock()
```

Comment:
Is this just a preference or have to do with the bug itself?

Reply:
it’s just a bit like an eye hurt )) that the mutex remains blocked after the fall
if try to catch an exception from outside with err := recover()